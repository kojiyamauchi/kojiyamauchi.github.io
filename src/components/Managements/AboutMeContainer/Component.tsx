/*
  About Me Container Component.
*/

import { useRouter } from 'next/router'
import { shallowEqual, useSelector } from 'react-redux'

import { AboutMe } from '@/components/Presentations/AboutMe'
import { Copyright } from '@/components/Presentations/Copyright'
import { store } from '@/ducks'
import { aboutMeToggle } from '@/ducks/AboutMe'
import { StoreTypes } from '@/types/type'

import styles from './style.module.scss'

export const closeAboutMe = (): void => void store.dispatch(aboutMeToggle({ isOpen: false }))

export const AboutMeContainer: React.VFC = (): JSX.Element => {
  const router = useRouter()
  const aboutMe = useSelector<StoreTypes, boolean>((state) => state.aboutMeStore.isOpen, shallowEqual)

  return (
    <div
      className={`${styles.wrapper} about-me-container ${aboutMe ? 'is-active' : ''}`}
      style={{ backgroundImage: `${router ? `url("${router.basePath}/webps/logo.webp")` : 'url("webps/logo.webp")'}` }}
    >
      <AboutMe closeAboutMe={closeAboutMe} />
      <div className={`${styles['copyright-wrapper']} copyright-wrapper`}>
        <Copyright className="copyright" parent="about-me" />
      </div>
    </div>
  )
}
