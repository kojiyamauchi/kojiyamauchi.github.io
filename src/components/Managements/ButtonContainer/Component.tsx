/*
  Button Container Component.
*/

import { shallowEqual, useSelector } from 'react-redux'

import { CloseButton } from '@/components/Presentations/CloseButton'
import { DynamicComponentOpenButton } from '@/components/Presentations/OpenButton'
import { store } from '@/ducks'
import { modalToggle } from '@/ducks/Modal'
import { StoreTypes } from '@/types/type'

import styles from './style.module.scss'

export const modalOpen = (): void => void store.dispatch(modalToggle({ isOpen: true }))
export const modalClose = (): void => void store.dispatch(modalToggle({ isOpen: false }))

type Props = {
  storybook?: boolean
}

export const ButtonContainer: React.VFC<Props> = ({ storybook = false }): JSX.Element => {
  const isOpen = useSelector<StoreTypes, boolean>((state) => state.modalStore.isOpen, shallowEqual)
  return (
    <div className={`${styles.wrapper} button-container`}>
      <DynamicComponentOpenButton modalState={isOpen} modalOpen={modalOpen} storybook={storybook} />
      <CloseButton modalState={isOpen} modalClose={modalClose} />
    </div>
  )
}
