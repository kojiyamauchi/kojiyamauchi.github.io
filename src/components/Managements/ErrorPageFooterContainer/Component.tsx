/*
  Error Page Footer Container Component.
*/

import { Copyright } from '@/components/Presentations/Copyright'
import styles from './style.module.scss'

export const ErrorPageFooterContainer: React.VFC = (): JSX.Element => {
  return (
    <footer className={`${styles.wrapper} error-page-footer-container`}>
      <Copyright parent="error" />
    </footer>
  )
}
