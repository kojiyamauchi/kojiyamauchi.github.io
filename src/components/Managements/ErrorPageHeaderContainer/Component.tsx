/*
  Error Page Header Container Component.
*/

import { ErrorPageHeading } from '@/components/Presentations/ErrorPageHeading'
import { ErrorPageNavigation } from '@/components/Presentations/ErrorPageNavigation'
import styles from './style.module.scss'

export const ErrorPageHeaderContainer: React.VFC = (): JSX.Element => {
  return (
    <header className={`${styles.wrapper} error-page-header-container`}>
      <ErrorPageHeading />
      <ErrorPageNavigation />
    </header>
  )
}
