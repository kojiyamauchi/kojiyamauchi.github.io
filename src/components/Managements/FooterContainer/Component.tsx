/*
  Footer Container Component.
*/

import { Copyright } from '@/components/Presentations/Copyright'

import styles from './style.module.scss'

export const FooterContainer: React.VFC = (): JSX.Element => {
  return (
    <footer className={`${styles.wrapper} footer-container`}>
      <Copyright parent="footer" />
    </footer>
  )
}
