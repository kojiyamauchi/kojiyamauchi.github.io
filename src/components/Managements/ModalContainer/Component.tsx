/*
  Modal Container Component.
*/

import { ResizeObserver } from '@juggle/resize-observer'
import { Canvas } from '@react-three/fiber'
import { Suspense } from 'react'

import { Copyright } from '@/components/Presentations/Copyright'
import { JumpToOldPortfolio } from '@/components/Presentations/JumpToOldPortfolio'
import { DynamicComponentLinkCube } from '@/components/Presentations/LinkCube'
import { ThanksForVisiting } from '@/components/Presentations/ThanksForVisiting'

import { useModal } from './Hooks'
import styles from './style.module.scss'

export const ModalContainer: React.VFC = (): JSX.Element => {
  const { myComponent, cubesProps, generatePositionValue, openAboutMe, copyrightHover } = useModal()

  return (
    <div className={`${styles.wrapper} modal-container`} ref={myComponent}>
      <div className={`${styles['graffiti-wrapper']} graffiti-wrapper`}>
        <ThanksForVisiting copyrightHover={copyrightHover} openAboutMe={openAboutMe} />
        <JumpToOldPortfolio copyrightHover={copyrightHover} openAboutMe={openAboutMe} />
      </div>
      <div className={`${styles['canvas-wrapper']} canvas-wrapper`}>
        <Canvas resize={{ polyfill: ResizeObserver }}>
          <Suspense fallback={null}>
            <directionalLight position={[1, 1, 1]} />
            <ambientLight intensity={0.65} />
            {cubesProps.map((cubeProps, index) => (
              <DynamicComponentLinkCube
                key={index}
                textureName={cubeProps.textureName}
                linkUrl={cubeProps.linkUrl}
                positionValue={generatePositionValue[index]}
                rotate={cubeProps.rotate}
              />
            ))}
          </Suspense>
        </Canvas>
      </div>
      <div className={`${styles['copyright-wrapper']} copyright-wrapper`}>
        <Copyright className="copyright" parent="modal" />
      </div>
    </div>
  )
}
