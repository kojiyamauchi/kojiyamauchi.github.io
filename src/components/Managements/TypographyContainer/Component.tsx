/*
  Typography Container Component.
*/

import { Typography } from '@/components/Presentations/Typography'
import { SubLetter } from '@/components/Presentations/SubLetter'
import { useTypography } from './Hooks'
import styles from './style.module.scss'

type Props = {
  visited: boolean
}

export const TypographyContainer: React.VFC<Props> = ({ visited }): JSX.Element => {
  const { myComponent } = useTypography(visited)

  return (
    <div className={`${styles.wrapper} typography-container`} ref={myComponent}>
      <Typography visited={visited} />
      <SubLetter visited={visited} />
    </div>
  )
}
