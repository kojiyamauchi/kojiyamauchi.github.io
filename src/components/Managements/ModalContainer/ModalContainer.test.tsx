/**
 * @jest-environment jsdom
 */
/*
  Modal Container Component Unit Test.
*/

import { ModalContainer } from './Component'
import { Provider } from 'react-redux'
import { store } from '@/ducks'
import { modalToggle } from '@/ducks/Modal'
import { render, act } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
// import userEvent from '@testing-library/user-event'
// import { renderHook } from '@testing-library/react-hooks/dom/'

describe('Modal Container Component Unit Test', () => {
  it('Snap Shot Testing', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <ModalContainer />
      </Provider>
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it('Operation Testing', () => {
    const { container } = render(
      <Provider store={store}>
        <ModalContainer />
      </Provider>
    )
    act(() => void store.dispatch(modalToggle({ isOpen: true })))
    expect(container.firstChild).toHaveClass('is-display')
    expect(container.firstChild).toHaveClass('is-active')
    act(() => void store.dispatch(modalToggle({ isOpen: false })))
    expect(container.firstChild).not.toHaveClass('is-active')
  })
})
