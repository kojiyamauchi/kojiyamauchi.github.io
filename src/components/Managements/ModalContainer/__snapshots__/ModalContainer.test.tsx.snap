// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Container Component Unit Test Snap Shot Testing 1`] = `
<DocumentFragment>
  <div
    class="wrapper modal-container is-display"
  >
    <div
      class="graffiti-wrapper graffiti-wrapper"
    >
      <div
        class="wrapper thanks-for-visiting "
      >
        <svg
          class="svg-graffiti svg-graffiti"
          enable-background="new 0 0 760 650;"
          version="1.1"
          viewBox="0 0 760 650"
          x="0px"
          xml:space="preserve"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
          y="0px"
        >
          <path
            class="path path"
            d="M461.7,621.7c4.1-87.7,21.6-174.7,51.5-257.1c6.6-18.2,14-36.5,25.6-51.9c11.5-15.2,26.9-27.3,44.4-34.8c10.1-4.3,21.2-7.1,32-5.4c14.6,2.3,27,12.7,35.2,25c8.2,12.4,12.7,26.7,17.2,40.8c4.2,13.3,8.5,26.7,12.7,40c10.3,32.4,20.6,64.9,27.9,98.1c12,54.2,16,110.1,11.9,165.5"
          />
          <path
            class="path path"
            d="M624.8,523c12.4,1.7,26.4,5.5,31.6,16.9c2.6,5.6,2.5,12,2.4,18.1c-0.1,5.9-0.3,11.8-0.4,17.7c-0.2,10.8-0.8,22.5-7.8,30.8c-3.5,4.1-8.3,6.9-13.2,9c-14,5.9-29.6,6.7-44.7,7.1c-17.2,0.4-34.6,0.4-51.6-2.3c-8.5-1.3-17.7-3.9-22.4-11c-1.8-2.8-2.8-6.1-3.7-9.3c-4.8-18.4-6.4-37.7-4.6-56.6c0.3-3,0.7-6.1,1.8-8.9c6.8-17.7,34.2-13.9,49-14.2C582.4,519.7,603.7,520.1,624.8,523z"
          />
          <path
            class="path path"
            d="M665.2,552.9c14.8-2.9,29.6-5.7,44.4-8.6"
          />
          <path
            class="path path"
            d="M662.2,589.2c16.6-1.2,33.3-0.7,49.8,1.5"
          />
          <path
            class="path path"
            d="M502.8,544.5c-9.6-3.8-19.6-6.9-29.7-9.2"
          />
          <path
            class="path path"
            d="M505.9,592.1c-12-2.1-24.1-3.5-36.3-4.1"
          />
          <path
            class="path path"
            d="M599.6,423.5c18.1-0.3,36.3-0.1,54.4,0.6"
          />
          <path
            class="path path"
            d="M515.3,420.9c16.8,0.3,33.6,0.6,50.3,0.8"
          />
          <path
            class="path path"
            d="M626,455.6c-0.3,6.2-0.2,12.5,0.2,18.7"
          />
          <path
            class="path path"
            d="M544,455c-0.2,5.3-0.2,10.6,0.2,15.8"
          />
          <path
            class="path path"
            d="M608.4,218c-1,15.8-1,31.6,0.1,47.4"
          />
          <path
            class="path path"
            d="M614.3,268.9c8.9-16,19.9-30.8,32.6-44"
          />
          <path
            class="path path"
            d="M601.5,270.5c-12.2-13.4-25.7-25.6-40.3-36.3"
          />
          <path
            class="path path"
            d="M417.2,550.6c11.9,21.4,22.7,43.5,32.4,66"
          />
          <path
            class="path path"
            d="M373.5,564c16.7,22.7,33.5,45.3,50.2,68"
          />
          <path
            class="path path"
            d="M415.2,549.4c9.2-11.3,16.3-24.3,20.9-38.1c2.4-7.3,3-17.6-4.2-20.6c-4-1.7-8.7,0-11.9,3s-4.9,7.1-6.7,11c-2.3-13-8-25.5-16.4-35.7c-0.9-1.1-1.9-2.3-3.3-2.6c-2.6-0.5-4.7,2-5.9,4.3c-3.6,6.8-5.1,14.8-4.1,22.5c-1.7-4.6-3.4-9.3-6.4-13c-3.1-3.8-7.8-6.6-12.6-6.1c-0.7,10.2-0.1,20.5,1.8,30.5c-5-7.8-13.8-13-23.1-13.7c-1.2-0.1-2.6,0-3.3,0.9c-0.7,0.8-0.6,2-0.4,3.1c1.5,9.7,4.6,19.2,9.1,28c-5-1.1-10.1-1.6-15.2-1.3c-1.6,0.1-3.4,0.3-4.6,1.4c-1.3,1.2-1.5,3.2-1.4,5c0.8,9.8,9.6,16.8,17.6,22.5c5.9,4.3,11.9,8.5,17.8,12.8"
          />
          <path
            class="path path"
            d="M432.9,473.4c5.8,2.5,10.6,7.3,13.2,13"
          />
          <path
            class="path path"
            d="M445.3,461.7c4.7,3.1,8.7,7.4,11.4,12.3"
          />
          <path
            class="path path"
            d="M313.6,514.4c-0.9,10.2,1.3,20.8,6.3,29.8"
          />
          <path
            class="path path"
            d="M303,525c-1.8,8.2-0.8,17,2.7,24.6"
          />
          <path
            class="path path"
            d="M555.6,552.4c16.8,9.4,32.7,20.6,47.3,33.1"
          />
          <path
            class="path path"
            d="M599.5,547.1c-15.2,11-29.4,23.5-42.1,37.3"
          />
          <path
            class="path path"
            d="M345.1,23.3c36.7,50.5,110,161.1,142.3,214.6"
          />
          <path
            class="path path"
            d="M437.1,419.7c-68.5,7.1-252.2,53.2-315.9,79.4"
          />
          <path
            class="path path"
            d="M11.6,16.4c18.8,0,37.6,0,56.4,0"
          />
          <path
            class="path path"
            d="M43.1,23.5c-1.2,33.4-0.2,66.9,2.9,100.2"
          />
          <path
            class="path path"
            d="M89,13.7c2.1,37.6,0,75.4-6.2,112.6c-0.4-9.8-0.3-19.9,3.4-29S98.2,80.4,108,79.7c3.4-0.2,6.9,0.4,9.6,2.5c4.7,3.5,5.9,9.9,6.4,15.8c1.2,13.7,0.7,27.5-1.6,41.1"
          />
          <path
            class="path path"
            d="M144.7,86.4c8.6-3,18-3.9,27-2.4c1.4,0.2,2.8,0.5,3.7,1.6c1.1,1.2,1.1,3.1,1.1,4.7c-0.4,13.7-0.8,27.3-1.2,41c-0.1,2.5-0.1,5.1,0.9,7.3c2.1,4.6,7.6,6.2,12.5,7.3"
          />
          <path
            class="path path"
            d="M170,110.4c-3.8-1.7-7.7-3.5-11.8-3.9s-8.7,0.7-11.4,3.8c-1.6,1.9-2.5,4.4-3.1,6.8c-1.4,5.6-1.8,11.4-1.2,17.2c0.4,3.6,1.2,7.3,3.5,10.1c5,6.2,14.3,5.9,22.2,5"
          />
          <path
            class="path path"
            d="M216.1,83.5c-1.2,22.9-3.4,45.8-6.7,68.6c-0.3-7.8-0.1-15.6,0.6-23.4c0.4-4.5,1-9,3-13s5.7-7.5,10.1-8.1c3-0.4,6.1,0.4,9.2,0.3c1.8-0.1,3.7-0.6,5.4,0.2c2,1,2.8,3.4,3.3,5.6c3,13,4.1,26.4,3.3,39.7"
          />
          <path
            class="path path"
            d="M269.3,52.2c1,34.9-0.5,69.9-4.4,104.6"
          />
          <path
            class="path path"
            d="M305.2,79.2c-11.4,6.6-21.8,15-30.7,24.7"
          />
          <path
            class="path path"
            d="M276.6,111.1c13.3,20.2,29.8,38.3,48.7,53.4"
          />
          <path
            class="path path"
            d="M248.1,190.6c-1.6-1.3-3.7-1.7-5.8-2c-3.5-0.6-7-1.2-10.5-0.6s-6.9,2.6-8.3,5.8c-0.9,2.1-0.8,4.4-0.7,6.6c0.7,23.2,2.5,46.3,5.2,69.3"
          />
          <path
            class="path path"
            d="M194.8,235.1c19-1.7,37.9-3.5,56.9-5.2"
          />
          <path
            class="path path"
            d="M277.1,253c-2.8-7.1-5.1-14.3-6.8-21.8c-0.3-1.2-0.5-2.5,0-3.7c0.6-1.4,2.1-2.1,3.4-2.6c5.4-2.2,11.2-3.6,17-4.4c1.9-0.2,3.9-0.4,5.6,0.5c2,1,3.1,3,4.1,5c2.9,6.2,4.8,12.8,5.6,19.6c0.1,1.3,0.2,2.6-0.3,3.8c-0.4,1-1.3,1.7-2.1,2.3c-7,5.5-15.9,8.6-24.8,8.8"
          />
          <path
            class="path path"
            d="M327.5,212.4c0.6,20.1-0.6,40.3-3.7,60.2c-1.7-10,4.2-19.5,9.7-28c4.1-6.2,8.5-12.6,15.1-15.8c4.6-2.2,9.8-2.6,14.8-2.9"
          />
          <path
            class="path path"
            d="M364.9,108.9c-7.3-3.7-17.2-2.7-22.5,3.6c-4.8,5.7-4.5,15,0.6,20.4c4.1,4.3,10.6,6.2,13.5,11.3c2.3,4,1.9,9,0,13.2s-5,7.7-8,11.1c-2.5,2.7-4.9,5.5-7.4,8.2"
          />
          <path
            class="path path"
            d="M16,321.3c23.7,40.7,49.2,80.5,76.4,119.2c16.5-38.5,28.5-78.5,35.7-118.9"
          />
          <path
            class="path path"
            d="M155,335.3c6.2,26,12.4,52,18.6,78c-1.1-0.6-2.1-1.2-3.2-1.9"
          />
          <path
            class="path path"
            d="M152.1,308.9c-1.1,0.3-2.2-1-1.7-2c1.8-1.4,4.8-1.4,6.4,0.1c-1.4,1.4-3.2,2.4-5.1,2.9"
          />
          <path
            class="path path"
            d="M209.3,333.2c-9.3,1.2-20.6-2.8-27,3.7c-4.8,4.9-2.9,12.2-0.8,18.2c2.9,7.9,6.3,16.3,14.1,20.4c6.1,3.1,13.6,2.9,20.4,4.5c6.8,1.6,13.7,6.5,12.5,13c-1.1,6.2-8.6,9.6-15.1,11.9c-4.4,1.5-8.8,3.1-13.2,4.6"
          />
          <path
            class="path path"
            d="M239.1,340.4c4.1,18.3,8.3,36.5,12.4,54.8"
          />
          <path
            class="path path"
            d="M234.1,312.1c-0.6-2.3-1.2-4.6-1.8-6.9"
          />
          <path
            class="path path"
            d="M255.7,326.9c11.4-0.7,22.7-2,34-3.8"
          />
          <path
            class="path path"
            d="M272.5,295.8c6,31.9,15.3,63.1,27.5,93.2"
          />
          <path
            class="path path"
            d="M320.9,335.7c1.9,15.1,4.4,30.1,7.3,45.1"
          />
          <path
            class="path path"
            d="M319.5,309.8c0.2,1,0.4,2,0.7,3c-2.7,1.1-6.1-0.9-6-3.5"
          />
          <path
            class="path path"
            d="M342.8,326.9c0.2,16.5,6.5,32.5,7,49.1c1.4-13.1-5.1-26.2-3.2-39.3c0.4-3.1,1.4-6.3,3.8-8.3c3.1-2.6,7.7-2.5,11.8-2.2c2.2,0.1,4.5,0.3,6.5,1.4c3.3,1.9,4.6,5.9,5.5,9.6c3.5,13.8,5.6,28,6.3,42.2"
          />
          <path
            class="path path"
            d="M425,295.2c-8.8-1.4-18.5,0.8-25.9,5.8c-2.6,1.8-5,4-6.4,6.6c-4.2,7.8,1.2,16.7,7.9,22.3c4.3,3.6,9.7,6.7,15.6,5.9c8.4-1.1,13.9-9.5,14.5-17.1c0.6-7.6-2.1-15-3-22.5c10.6,15.3,15.4,33.4,20,51.1c1.1,4.3,2.2,9.1,0.3,13.3c-2.2,4.8-7.7,7.7-12.9,10.1c-10.3,4.7-20.9,8.7-31.8,12"
          />
        </svg>
        <button
          class="button-about-me button-about-me"
        >
          About Me üèÑ‚Äç‚ôÇÔ∏è
        </button>
      </div>
      <div
        class="wrapper jump-to-old-portfolio  "
      >
        <svg
          class="svg-graffiti svg-graffiti"
          enable-background="new 0 0 760 650;"
          version="1.1"
          viewBox="0 0 760 650"
          x="0px"
          xml:space="preserve"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
          y="0px"
        >
          <path
            class="path path"
            d="M461.7,621.7c4.1-87.7,21.6-174.7,51.5-257.1c6.6-18.2,14-36.5,25.6-51.9c11.5-15.2,26.9-27.3,44.4-34.8c10.1-4.3,21.2-7.1,32-5.4c14.6,2.3,27,12.7,35.2,25c8.2,12.4,12.7,26.7,17.2,40.8c4.2,13.3,8.5,26.7,12.7,40c10.3,32.4,20.6,64.9,27.9,98.1c12,54.2,16,110.1,11.9,165.5"
          />
          <path
            class="path path"
            d="M624.8,523c12.4,1.7,26.4,5.5,31.6,16.9c2.6,5.6,2.5,12,2.4,18.1c-0.1,5.9-0.3,11.8-0.4,17.7c-0.2,10.8-0.8,22.5-7.8,30.8c-3.5,4.1-8.3,6.9-13.2,9c-14,5.9-29.6,6.7-44.7,7.1c-17.2,0.4-34.6,0.4-51.6-2.3c-8.5-1.3-17.7-3.9-22.4-11c-1.8-2.8-2.8-6.1-3.7-9.3c-4.8-18.4-6.4-37.7-4.6-56.6c0.3-3,0.7-6.1,1.8-8.9c6.8-17.7,34.2-13.9,49-14.2C582.4,519.7,603.7,520.1,624.8,523z"
          />
          <path
            class="path path"
            d="M665.2,552.9c14.8-2.9,29.6-5.7,44.4-8.6"
          />
          <path
            class="path path"
            d="M662.2,589.2c16.6-1.2,33.3-0.7,49.8,1.5"
          />
          <path
            class="path path"
            d="M502.8,544.5c-9.6-3.8-19.6-6.9-29.7-9.2"
          />
          <path
            class="path path"
            d="M505.9,592.1c-12-2.1-24.1-3.5-36.3-4.1"
          />
          <path
            class="path path"
            d="M599.6,423.5c18.1-0.3,36.3-0.1,54.4,0.6"
          />
          <path
            class="path path"
            d="M515.3,420.9c16.8,0.3,33.6,0.6,50.3,0.8"
          />
          <path
            class="path path"
            d="M626,455.6c-0.3,6.2-0.2,12.5,0.2,18.7"
          />
          <path
            class="path path"
            d="M544,455c-0.2,5.3-0.2,10.6,0.2,15.8"
          />
          <path
            class="path path"
            d="M608.4,218c-1,15.8-1,31.6,0.1,47.4"
          />
          <path
            class="path path"
            d="M614.3,268.9c8.9-16,19.9-30.8,32.6-44"
          />
          <path
            class="path path"
            d="M601.5,270.5c-12.2-13.4-25.7-25.6-40.3-36.3"
          />
          <path
            class="path path"
            d="M417.2,550.6c11.9,21.4,22.7,43.5,32.4,66"
          />
          <path
            class="path path"
            d="M373.5,564c16.7,22.7,33.5,45.3,50.2,68"
          />
          <path
            class="path path"
            d="M415.2,549.4c9.2-11.3,16.3-24.3,20.9-38.1c2.4-7.3,3-17.6-4.2-20.6c-4-1.7-8.7,0-11.9,3s-4.9,7.1-6.7,11c-2.3-13-8-25.5-16.4-35.7c-0.9-1.1-1.9-2.3-3.3-2.6c-2.6-0.5-4.7,2-5.9,4.3c-3.6,6.8-5.1,14.8-4.1,22.5c-1.7-4.6-3.4-9.3-6.4-13c-3.1-3.8-7.8-6.6-12.6-6.1c-0.7,10.2-0.1,20.5,1.8,30.5c-5-7.8-13.8-13-23.1-13.7c-1.2-0.1-2.6,0-3.3,0.9c-0.7,0.8-0.6,2-0.4,3.1c1.5,9.7,4.6,19.2,9.1,28c-5-1.1-10.1-1.6-15.2-1.3c-1.6,0.1-3.4,0.3-4.6,1.4c-1.3,1.2-1.5,3.2-1.4,5c0.8,9.8,9.6,16.8,17.6,22.5c5.9,4.3,11.9,8.5,17.8,12.8"
          />
          <path
            class="path path"
            d="M432.9,473.4c5.8,2.5,10.6,7.3,13.2,13"
          />
          <path
            class="path path"
            d="M445.3,461.7c4.7,3.1,8.7,7.4,11.4,12.3"
          />
          <path
            class="path path"
            d="M313.6,514.4c-0.9,10.2,1.3,20.8,6.3,29.8"
          />
          <path
            class="path path"
            d="M303,525c-1.8,8.2-0.8,17,2.7,24.6"
          />
          <path
            class="path path"
            d="M555.6,552.4c16.8,9.4,32.7,20.6,47.3,33.1"
          />
          <path
            class="path path"
            d="M599.5,547.1c-15.2,11-29.4,23.5-42.1,37.3"
          />
          <path
            class="path path"
            d="M345.1,23.3c36.7,50.5,110,161.1,142.3,214.6"
          />
          <path
            class="path path"
            d="M437.1,419.7c-68.5,7.1-252.2,53.2-315.9,79.4"
          />
          <path
            class="path path"
            d="M36.6,17.8c20.1,2,40.4,2.8,60.6,2.3"
          />
          <path
            class="path path"
            d="M64.6,27.3c3.9,21.6,4.7,43.8,2.1,65.6c-0.8,6.7-1.9,13.5-5.4,19.3c-3.5,5.8-9.8,10.2-16.5,9.7c-5-0.4-9.5-3.5-12.7-7.5s-5-8.8-6.7-13.5c-2.8-7.8-5.1-15.7-6.9-23.7"
          />
          <path
            class="path path"
            d="M89.5,78.9c-0.7,9.8-0.6,19.7,0.5,29.5c0.6,5.7,1.6,11.5,5,16.1c3.7,5.1,10.2,7.9,16.5,7.3c1.7-0.2,3.4-0.6,4.5-1.8c0.8-0.9,1.3-2.1,1.6-3.3c4.1-13.1,2.6-27.2,1.1-40.8"
          />
          <path
            class="path path"
            d="M138.5,89.5c0,14.4,0,28.8,0,43.2c0.3-10.1,0.6-20.2,0.9-30.3c0.1-3.4,0.3-7.1,2.6-9.7c2.9-3.3,7.8-3.4,12.1-3.3c1.4,0,2.9,0.1,3.8,1.2c0.6,0.7,0.8,1.6,0.9,2.5c2,12.4,3.2,24.9,3.5,37.5c-0.1-10.2-0.1-20.3-0.2-30.5c0-3.9,0-8,2-11.3c3.7-0.1,7.3-0.1,11-0.2c0.9,0,1.9,0,2.6,0.5c0.6,0.4,0.9,1.1,1.1,1.8c4.9,12.1,7.7,25.1,8.1,38.2"
          />
          <path
            class="path path"
            d="M203.2,78.1c1,26.9,2,53.8,3,80.8"
          />
          <path
            class="path path"
            d="M207.9,62.3c0.5-4.2,5.6-6,9.8-6.8c4.1-0.8,8.2-1.6,12.3-0.8c6.1,1.2,11.2,5.9,14,11.5c2.8,5.6,3.6,12,3.7,18.2c0,4.2-0.3,8.6-2.3,12.3c-2.1,4-5.8,6.8-9.5,9.3c-6.8,4.5-14.1,8.3-21.7,11.1"
          />
          <path
            class="path path"
            d="M273.7,143.9c3.5,0,7,0,10.5,0c6.4,0,12.9,0,19.3,0"
          />
          <path
            class="path path"
            d="M283.3,117.6c1.8,21.3,2.9,42.7,3.3,64.1"
          />
          <path
            class="path path"
            d="M325.3,182c-4-3.4-7.1-8-8.7-13c-0.7-2.2-1.1-4.5-0.6-6.8c1.4-5.7,8.1-8.1,13.8-9.1c3.7-0.6,7.6-1,10.9,0.8c3.1,1.7,5,5.2,5.6,8.7c0.6,3.5,0.2,7.1-0.2,10.7c-0.3,2.5-0.8,5.3-2.8,6.7c-1.1,0.8-2.5,1-3.9,1.1c-4.6,0.5-9.3,0.7-13.9,0.6"
          />
          <path
            class="path path"
            d="M41.6,203.5c1.5-0.7,3.1-1,4.7-1.3c12.8-2.2,26-4.1,38.7-1.6c3.1,0.6,6.3,1.6,8.7,3.6c4.2,3.6,5.2,9.6,6,15c1.7,12.4,8.8,32.2-3.6,40.5c-9,6.1-26.3,8.7-36.9,7c-11.2-1.7-18.7-8.6-22-19.3C34.4,238.1,30.5,209.2,41.6,203.5z"
          />
          <path
            class="path path"
            d="M141.4,191.4c-5.4,25.3-7.1,51.5-5,77.3"
          />
          <path
            class="path path"
            d="M211.7,187c-1,27.5-2.3,55.1-3.8,82.6"
          />
          <path
            class="path path"
            d="M199.5,226.3c-5.9-2.3-12.4-4.7-18.5-2.8c-8.2,2.6-12,12-13.5,20.4c-1.5,8.7-1.4,18,2.8,25.7c4.2,7.7,13.5,13.1,21.9,10.7"
          />
          <path
            class="path path"
            d="M59.6,327.3c-0.2,36.5-3.3,73-9.4,109"
          />
          <path
            class="path path"
            d="M64.2,315.9c9.1-4.1,19.5-3.9,29.5-3.6c2.7,0.1,5.5,0.2,8,1.1c7.4,2.8,10.3,11.5,11.6,19.3c1,5.9,1.5,12,0.5,18c-1.1,5.9-3.9,11.7-8.7,15.4c-3.2,2.5-7.1,3.8-10.9,5.2c-7.4,2.6-14.8,5.2-22.2,7.8"
          />
          <path
            class="path path"
            d="M116.8,403.8c0.2-5.4,0.5-11.4,4-15.5c4.4-5.2,12.2-5.7,19-5.8c2.9,0,5.9-0.1,8.6,1.2c4.6,2.2,6.7,7.6,8.1,12.5c2.3,7.8,4.2,15.9,3.5,23.9c-0.2,1.8-0.5,3.7-1.5,5.2c-1.8,2.6-5.2,3.5-8.3,4.2c-7.9,1.8-20.7,7.8-27.8,1.8C115.8,425.7,116.6,411.5,116.8,403.8z"
          />
          <path
            class="path path"
            d="M171.9,379.2c2,17.1,3,34.3,3.1,51.5c0-10.3-0.1-20.6-0.1-30.8c0-3.3,0-6.8,1.7-9.7c3.2-5.5,10.5-6.4,16.8-6.7"
          />
          <path
            class="path path"
            d="M199.5,358.9c10.2,1,20.5,1.9,30.7,2.9"
          />
          <path
            class="path path"
            d="M213.8,329.2c2.7,31.6,6,63.2,10,94.7"
          />
          <path
            class="path path"
            d="M290.5,345.9c-2.5-7.2-6.2-14-11-20c-4.7-5.9-12.3-11.3-19.2-8.4c-6.1,2.6-8.1,10.1-8.6,16.7c-2,27.3,8.8,54.5,6.1,81.8"
          />
          <path
            class="path path"
            d="M236.3,385.8c12.3-0.5,24.6-1,36.9-1.6"
          />
          <path
            class="path path"
            d="M300.4,401.5c-4.6-5.5-6.2-13.4-4-20.2c0.6-1.8,1.5-3.6,3-4.7c1.9-1.4,4.4-1.6,6.8-1.6c5.7-0.2,11.5-0.3,17.2-0.3c1.8,0,3.6,0,5.2,0.6c4.1,1.6,5.9,6.6,5.8,11.1C334.2,405,314,417.5,300.4,401.5z"
          />
          <path
            class="path path"
            d="M349.9,302.8c3.4,30.6,6.1,61.3,8,92.1"
          />
          <path
            class="path path"
            d="M381.3,360.7c-0.4,10.5-0.3,21.1,0.3,31.6"
          />
          <path
            class="path path"
            d="M379.5,331.7c0-1.1,0.1-2.2,0.1-3.4c1.5-0.5,3.2-0.3,4.5,0.7c-0.8,1.3-3,1.3-3.8,0.1"
          />
          <path
            class="path path"
            d="M405.9,372.1c-2.5-6-3.8-12.5-3.8-19c0-4.3,1-9.4,4.9-11.3c1.5-0.7,3.2-0.9,4.8-1.1c6.6-0.6,13.9-1.1,19.4,2.7c7.4,5.2,8.1,15.6,8.1,24.6c0,3-0.1,6.2-2.1,8.4c-1.9,2.1-14.6,7.3-17.6,7.4C412.8,384.1,408.4,378,405.9,372.1z"
          />
        </svg>
        <button
          class="button-about-me button-about-me"
        >
          About Me üèÑ‚Äç‚ôÄÔ∏è
        </button>
      </div>
    </div>
    <div
      class="canvas-wrapper canvas-wrapper"
    >
      <div
        style="position: relative; width: 100%; height: 100%; overflow: hidden;"
      >
        <canvas
          style="display: block;"
        />
      </div>
    </div>
    <div
      class="copyright-wrapper copyright-wrapper"
    >
      <a
        class="wrapper copyright is-modal"
        href="http://kojiyamauchi.com"
        rel="noreferrer"
        target="_blank"
      >
        ¬© 2021 All The Small Things, All Rights Reserved.
      </a>
    </div>
  </div>
</DocumentFragment>
`;
